apps/dav/lib/comments/commentsplugin.php

```
@@ -0,0 +1,243 @@
+class CommentsPlugin extends ServerPlugin {
+	// namespace
+	const NS_OWNCLOUD = 'http://owncloud.org/ns';
+
+	const REPORT_PARAM_LIMIT     = '{http://owncloud.org/ns}limit';
+	const REPORT_PARAM_OFFSET    = '{http://owncloud.org/ns}offset';
+	const REPORT_PARAM_TIMESTAMP = '{http://owncloud.org/ns}datetime';
+
+	/** @var ICommentsManager  */
+	protected $commentsManager;
+
+	/** @var \Sabre\DAV\Server $server */
+	private $server;
+
+	/** @var  \OCP\IUserSession */
+	protected $userSession;
+
+	/**
+	 * Comments plugin
+	 *
+	 * @param ICommentsManager $commentsManager
+	 * @param IUserSession $userSession
+	 */
+	public function __construct(ICommentsManager $commentsManager, IUserSession $userSession) {
+		$this->commentsManager = $commentsManager;
+		$this->userSession = $userSession;
+	}
+
+	/**
+	 * This initializes the plugin.
+	 *
+	 * This function is called by Sabre\DAV\Server, after
+	 * addPlugin is called.
+	 *
+	 * This method should set up the required event subscriptions.
+	 *
+	 * @param Server $server
+	 * @return void
+	 */
+	function initialize(Server $server) {
+		$this->server = $server;
+		if(strpos($this->server->getRequestUri(), 'comments/') !== 0) {
+			return;
+		}
+
+		$this->server->xml->namespaceMap[self::NS_OWNCLOUD] = 'oc';
+
+		$this->server->xml->classMap['DateTime'] = function(Writer $writer, \DateTime $value) {
+			$writer->write($value->format('Y-m-d H:m:i'));
+		};
+
+		$this->server->on('report', [$this, 'onReport']);
+		$this->server->on('method:POST', [$this, 'httpPost']);
+	}
+
+	/**
+	 * POST operation on Comments collections
+	 *
+	 * @param RequestInterface $request request object
+	 * @param ResponseInterface $response response object
+	 * @return null|false
+	 */
+	public function httpPost(RequestInterface $request, ResponseInterface $response) {
+		$path = $request->getPath();
+
+		// Making sure the node exists
+		try {
+			$node = $this->server->tree->getNodeForPath($path);
+		} catch (NotFound $e) {
+			return null;
```

If the node doesn't exist you should let the exception be thrown.
Basically just remove the try/catch block.